### ISSUE TITLE
quodlibet wakes up frequently even when paused

### ISSUE THREAD

**Issue by joshtriplett** [quodlibet/quodlibet#2757]
With quodlibet running but paused, I get the following in powertop:

```
             10.8 ms/s      57.3        Process        /usr/bin/python /usr/bin/quodlibet --run --play-file
```

The exact amount varies, but on average, quodlibet is waking up more than 50 times per second.

Just to make sure, I tried disabling all plugins, and the wakeups remained.

I straced quodlibet, and after it reached a steady state, it appeared to regularly call `poll` with a timeout value of 17 ms, which seems consistent with the approximate number of wakeups.

**Comment by joshtriplett** [comment]
Interestingly, I'm seeing similar behavior (a repeated poll call with a timeout around 17ms) from gnome-terminal-server and gnome-control-center as well. Trying to track it down.

**Comment by lazka** [comment]
I can't reproduce.

There are some problems currently with starting QL by opening a file with it instead of simply starting it. Maybe there is a difference.

Are you on wayland maybe?

**Comment by joshtriplett** [comment]
Yes, I'm on Wayland. I'm currently trying to track down the problem. Current hypothesis: I think there may be some issue with GTK images or animations waking up every frame to redraw even when unchanged.

**Comment by joshtriplett** [comment]
Just confirmed that hypothesis. So, this isn't a bug in quodlibet, it's a bug in GTK.

**Comment by joshtriplett** [comment]
Hmmm. I haven't yet managed to figure out *what* in quodlibet is keeping the GTK frame clock set at 60 fps. I tried setting the entire main window to visible=FALSE, and the frame clock stayed at 60fps. I don't know what is forcing that, or whether the bug is in GTK or quodlibet.

**Comment by lazka** [comment]
Is there any difference with a minimal example https://pygobject.readthedocs.io/en/latest/getting_started.html ?

**Comment by joshtriplett** [comment]
The minimal example doesn't wake up repeatedly, and doesn't have the frame clock running.

**Comment by lazka** [comment]
OK, let's investigate. Breaking on gdk_frame_clock_begin_updating might give some clue

**Comment by lazka** [comment]
I can't reproduce under wayland either. Have you tried running with "--no-plugins"?

**Comment by joshtriplett** [comment]
Yes, it reproduces with `quodlibet --no-plugins`.

**Comment by lazka** [comment]
no idea then

**Comment by lazka** [comment]
Any news?

**Comment by joshtriplett** [comment]
The GTK+ folks passed along that the current architecture of the frame clock is such that if you have a widget that's currently not being shown to the user, but whose size is being taken into account, such as a widget in a not-currently-shown tab of a notebook, or in a stack, then it still gets animated.

While they agreed that that was a problem, they also noted that it seems hard to fix. That's the last status update I have from them.

**Comment by lazka** [comment]
Thanks for the update. Still weird as we don't really animate anything, hidden or not, at least not on purpose. Last guess as to why I can't reproduce would be that it depends on the active QL browser.

**Comment by CreamyCookie** [comment]
@joshtriplett Thanks for the report. Do you know, or would you mind checking if this is still an issue in Quod Libet 4.4?
